lib/archethic/self_repair/notifier.ex

```diff
@@ -105,7 +119,132 @@ defmodule Archethic.SelfRepair.Notifier do
+      |> Stream.filter(&Utils.key_in_node_list?(elem(&1, 1), unavailable_node_key))
+      |> Stream.filter(&(&1 |> elem(1) != []))
+      |> Stream.map(
+        &sync_unavailable_summary_aggregate(&1, current_node_public_key, unavailable_node_key)
+      )
+      |> Stream.run()
+    end
+
+    :ok
+  end
+
+  @spec summary_aggregates_to_sync(summary_time :: DateTime.t()) ::
+          {summary_time :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())}
+  def summary_aggregates_to_sync(summary_time) do
+    node_list =
+      P2P.list_nodes()
+      |> Enum.filter(fn node ->
+        node.authorization_date != nil &&
+          DateTime.compare(node.authorization_date, summary_time) == :lt
+      end)
+
+    previous_chain_storage_nodes =
+      summary_time
+      |> Crypto.derive_beacon_aggregate_address()
+      |> Election.chain_storage_nodes(node_list)
+
+    {summary_time, previous_chain_storage_nodes}
+  end
+
+  @spec sync_unavailable_summary_aggregate(
+          {summary_time :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())},
+          current_node_public_key :: Crypto.key(),
+          unavailable_node_key :: Crypto.key()
+        ) :: :ok
+  def sync_unavailable_summary_aggregate(
+        {summary_time, previous_chain_storage_nodes},
+        current_node_public_key,
+        unavailable_node_key
+      ) do
+    new_chain_storage_nodes =
+      Election.chain_storage_nodes(
+        Crypto.derive_beacon_aggregate_address(summary_time),
+        P2P.authorized_nodes()
+        |> Enum.reject(&(&1.first_public_key == unavailable_node_key))
+      ) -- previous_chain_storage_nodes
+
+    with {:empty, false} <- {:empty, Enum.empty?(new_chain_storage_nodes)},
+         # current node in new_chain_storage_nodes but not in previous_chain_storage_nodes
+         {:prev_election, false} <-
+           {:prev_election,
+            Utils.key_in_node_list?(previous_chain_storage_nodes, current_node_public_key)},
+         {:new_election, true} <-
+           {:new_election,
+            Utils.key_in_node_list?(new_chain_storage_nodes, current_node_public_key)},
+         storage_nodes when storage_nodes != [] <-
+           Enum.filter(previous_chain_storage_nodes, &(&1.available? == true)),
+         {:ok, aggregate = %SummaryAggregate{}} <-
+           BeaconChain.fetch_summaries_aggregate_from_nodes(summary_time, storage_nodes) do
+      BeaconChain.write_summaries_aggregate(aggregate)
+
+      Logger.debug(
+        "Fetched and Stored Missing #{summary_time} Aggregate due to shard unavailablity",
+        Repair: "SelfRepair.Notifier"
+      )
+
+      :ok
+    else
+      {:empty, false} ->
+        Logger.debug("Error during Aggregate Repair",
+          Repair: "SelfRepair.Notifier",
+          error:
+            "There are currently no new nodes in the New (Election)/(shard selection) for Chain Storage Nodes. and Summarytime #{summary_time}"
+        )
+
+      {:prev_election, false} ->
```

Comment:
This case will not be handled as the `with` already handle the `false` value.
But the `true` case is not matched and will go on the latest case.